core/src/main/java/org/apache/iceberg/TableUtil.java

```diff
@@ -60,4 +61,28 @@ public static String metadataFileLocation(Table table) {
               "%s does not have a metadata file location", table.getClass().getSimpleName()));
     }
   }
+
+  public static boolean supportsRowLineage(Table table) {
+    Preconditions.checkArgument(null != table, "Invalid table: null");
+    if (table instanceof BaseMetadataTable) {
+      return false;
+    }
+
+    return formatVersion(table) >= TableMetadata.MIN_FORMAT_VERSION_ROW_LINEAGE;
+  }
+
+  public static Schema schemaWithRowLineage(Table table) {
```

Comment:
Should this be in `TableUtil`? It's a good util method, but it seems to fit better in `MetadataColumns` to me. This class just has work-arounds for things that aren't exposed through the `Table` API like version.

Reply:
I agree TableUtil probably isn't the rigt place to keep this method  MetadataColumns.withRowLineageSchema(Table table) is better.